lib/arrays.md

```diff
@@ -5,44 +5,59 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+I can think of one way to do it, that would involve two commands, the delete_at and insert commands.  I'm going to call this array names. You could write 
+names.delete_at(1)
+names.insert(1, "Florence")
+puts names.print
+
+This should print out ["Laura," "Florence", "Tori"]
 
 # What do the methods `push`, `pop`, `shift`, and `unshift` do?
 
-Replace me with your response...
+The method push adds an item to the end of the array.  Pop is a method used to remove an element from an array.  It can be used by itself to remove the last element from the array, or a particular element or elements can be specified with its associated integer and removed.  
+The shift method removes the first element from the array.  It is similar to pop, in that it removes elements, but if a number is specified, say (3), it will remove the first three elements from the array, or return the default (nil if not specified) if there is nothing in the array for what has been identified to be removed with shift.  The unshift method does the opposite of shift, and adds elements to the front of the array.  It does not delete elements that were in the front, but moves them down in the array to later/higher associated integers.
 
 # How do you determine how many elements are in an Array?
 
-Replace me with your response...
+The count method returns how many elements there are in the array.
 
 # How would you reverse the order of elements in an Array?
 
-Replace me with your response...
+Fittingly, the reverse method would return the elements in reverse order.   It would look like [ "j", "k", "l" ].reverse   #=> ["l", "k", "j"]
 
 # How would you convert a String `"Sumeet Jain"` into an Array `["Sumeet", "Jain"]`? How about to `["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]`? How would you convert the Array back into a String?
 
-Replace me with your response...
\ No newline at end of file
+To convert a string to an array, the split method is used.  I read that the default for split uses a space character to separate strings into elements for arrays.  I hope that is the case!  To convert "Sumeet Jain"` into an Array `["Sumeet", "Jain"]` I would assign the string "Sumeet Jain" a name, sumeet = "Sumeet Jain", and the method would look like sumeet.split .
+To convert "Sumeet Jain" into ["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"], I would specify a space, so it would be sumeet.split "" .
+To convert the array back to a string, I would use the join method.  I would assign a name for the array jain = ["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"].
+I would write sumeet_jain = jain.join 
+puts sumeet_jain
+I guess I might need to add a space in?  I could, after printing, add a space between the different points in that string...
```

Comment:
If your array is something like `sumeet_chars = ["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]`, `sumeet_chars.join` will include that space that's in the array at index `6`!

If your array is `sumeet = ["Sumeet", "Jain"]` you will have to do something a little different when you use `join`. The [documentation of that method](http://ruby-doc.org/core-1.9.3/Array.html#method-i-join) might help here!